contracts/protocol/facets/SellerHandlerFacet.sol

```diff
@@ -94,73 +94,176 @@ contract SellerHandlerFacet is SellerBase {
-        delete lookups.sellerIdByAdmin[seller.admin];
-        delete lookups.sellerIdByClerk[seller.clerk];
-        delete lookups.sellerIdByAuthToken[authToken.tokenType][authToken.tokenId];
-        delete protocolEntities().authTokens[seller.id];
+        if (seller.treasury != _seller.treasury) {
+            require(_seller.treasury != address(0), INVALID_ADDRESS);
+            // Update treasury
+            seller.treasury = _seller.treasury;
+            emitAppliedEvent = true;
+        }
 
-        // Store this address of existing seller operator to check if you have to transfer the ownership later
-        address oldSellerOperator = seller.operator;
+        if (_seller.operator != seller.operator) {
+            require(_seller.operator != address(0), INVALID_ADDRESS);
+            // Operator address owner must approve the update for prevent front-running
+            sellerUpdatePending.operator = _seller.operator;
+            needsApproval = true;
+        }
 
-        // Ignore active flag passed in by caller and set to value in storage.
-        _seller.active = seller.active;
-        storeSeller(_seller, _authToken);
+        if (_seller.clerk != seller.clerk) {
+            require(_seller.clerk != address(0), INVALID_ADDRESS);
+            // Clerk address owner must approve the update for prevent front-running
+            sellerUpdatePending.clerk = _seller.clerk;
+            needsApproval = true;
+        }
 
-        // If operator changed, transfer the ownership of NFT voucher
-        if (oldSellerOperator != _seller.operator) {
-            IBosonVoucher(lookups.cloneAddress[seller.id]).transferOwnership(_seller.operator);
+        if (needsApproval) {
+            emit SellerUpdatePending(_seller.id, sellerUpdatePending, sender);
+        }
+
+        if (emitAppliedEvent) {
+            // Notify watchers of state change
+            emit SellerUpdateApplied(_seller.id, seller, sellerUpdatePending, authToken, sender);
+        }
+    }
+
+    function optInToSellerUpdate(uint256 _sellerId) external {
+        // Cache protocol lookups for reference
+        ProtocolLib.ProtocolLookups storage lookups = protocolLookups();
+        address sender = msgSender();
+
+        // Get seller pending update
+        Seller storage _sellerPendingUpdate = lookups.sellerPendingUpdates[_sellerId];
+
+        require(
+            _sellerPendingUpdate.admin != address(0) ||
+                _sellerPendingUpdate.operator != address(0) ||
+                _sellerPendingUpdate.clerk != address(0),
+            NO_PENDING_SELLER_ROLE_UPDATE
+        );
+
+        preUpdateSellerCheck(_sellerId, sender, lookups);
+
+        // Get storage location for seller
+        (, Seller storage seller, AuthToken storage authToken) = fetchSeller(_sellerId);
+
+        // Approve operator address
+        if (_sellerPendingUpdate.operator == sender && _sellerPendingUpdate.operator != address(0)) {
```

Comment:
Me again on this line :)

 If `_sellerPendingUpdate.operator == sender` is true, then well, the second part `_sellerPendingUpdate.operator != address(0)` is also always true (since `sender != address(0)`).
 And if `_sellerPendingUpdate.operator == sender` is false, then second part is not even checked.
 
 So ultimately what you need is only the first part of the check before `&&`.

Sorry if my first comment about this line misled you.

Reply:
done